CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ option(BUILD_LIBSRTP_DESTINATION_PLATFORM "If buildng LibSRTP what is the destin
 option(BUILD_SAMPLE "Build available samples" ON)
 option(ENABLE_DATA_CHANNEL "Enable support for data channel" ON)
 option(ENABLE_KVS_THREADPOOL "Enable support for KVS thread pool in signaling" ON)
+option(ENABLE_SENDING_METRICS_TO_VIEWER "Enable sending master-side metrics to the viewer as a json string via the datachannel" OFF)
```

Comment:
This was discussed and agreed upon with James since this change should not work unless this flag is enabled as it is only intended for customers who want to benchmark. If this flag is enabled, the C SDK only accepts json messages from the other peer, attaches timestamps to them and responds with the timestamps from the signaling, ice agent, gathering phases

Reply:
I agree that the changes in this PR for sending profiling data via data channel should not happen unless an option is enabled. But, IMO it shouldnt be a CMake option given the core of the logic is in the callback that is implemented in the samples. CMake flags should be reserved for core SDK and should not be used for samples unless there is no other way to achieve the functionality. Is there a reason we cannot achieve setting this option by having a bool flag in one of the sample structs and pass that in as customData to `onDataChannelMessage`?

Reply:
Removed as discussed